src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs

```diff
@@ -112,13 +112,6 @@ private static void VerifyAction<TContext>(Action<TContext> action)
             {
                 throw new ArgumentNullException(nameof(action));
             }
-
-            // Disallow async methods to be registered.
-            // Suppression due to bug fixed in .NET 5: https://github.com/dotnet/runtime/issues/30968
-            if (action.GetMethodInfo()!.IsDefined(typeof(AsyncStateMachineAttribute)))
```

Comment:
Can we file an issue on roslyn-analyzers repo to author an RS rule in MS.CA.Analyzers for this?

Reply:
Filed https://github.com/dotnet/roslyn-analyzers/issues/4944